samples/subsys/zbus/benchmark/src/benchmark.c

```diff
@@ -183,7 +190,11 @@ static void producer_thread(void)
 
 	zbus_chan_finish(&bm_channel);
 
-	uint32_t start = k_uptime_get_32();
+#if defined(CONFIG_ARCH_POSIX)
+	uint32_t start_ns = native_rtc_gettime_us(RTC_CLOCK_PSEUDOHOSTREALTIME) * NSEC_PER_USEC;
+#else
+	uint32_t start_ns = k_cyc_to_ns_near32(sys_clock_cycle_get_32());
+#endif
```

Comment:
I would not use the declaration inside the ifdef. 

Suggestion:
```c
#if defined(CONFIG_ARCH_POSIX)
#define GET_ARCH_START_TIME (native_rtc_gettime_us(RTC_CLOCK_PSEUDOHOSTREALTIME) * NSEC_PER_USEC)
#else
#define GET_ARCH_START_TIME (k_cyc_to_ns_near32(sys_clock_cycle_get_32()))
#endif

uint32_t start_ns = GET_ARCH_START_TIME;
```

Reply:
Beautiful.
I called it `GET_ARCH_TIME_NS()` to also imply function calls are being made. That ok?

Reply:
Better